portal/src/main/webapp/WEB-INF/jsp/igv.jsp

```diff
@@ -35,60 +35,66 @@
 <%@ page import="java.net.URLEncoder" %>
 <%@ page import="com.google.common.base.Joiner" %>
 <%@ page import="java.io.UnsupportedEncodingException" %>
-<%@ page import="org.mskcc.cbio.portal.model.GeneWithScore" %>
-<%@ page import="org.mskcc.cbio.portal.servlet.QueryBuilder" %>
-<%@ page import="org.mskcc.cbio.portal.util.IGVLinking" %>
-<%@ page import="org.mskcc.cbio.portal.dao.DaoGeneOptimized" %>
-<%@ page import="org.mskcc.cbio.portal.model.CanonicalGene" %>
 
 <div class="section" id="igv_tab">
-    <table>
-        <tr>
-            <td style="padding-right:25px; vertical-align:top;"><img src="images/IGVlogo.png" alt=""/></td>
-            <td style="vertical-align:top">
+<script type="text/javascript">
+    $(document).ready(function(){
+        function generateHTML(cancerStudyId, hugoSymbol, sampleIds){
+            var headerContent = '<head>    <link rel="stylesheet" type="text/css"  href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/themes/smoothness/jquery-ui.css"/>'
+                + '<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css">'
+                + '<link rel="stylesheet" type="text/css" href="css/igv.css">'
+                + '<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"><\/script>'
+                + '<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/jquery-ui.min.js"><\/script>'
+                + '<script type="text/javascript" src="js/lib/igv.js"><\/script></head>';
+            var bodyContent1 = '<body><div id="igvDiv" style="padding-top: 10px;padding-bottom: 10px; border:1px solid lightgray;width:100%"></div><script type="text/javascript">  $(document).ready(function () {    var div = $("#igvDiv")[0],   options = {'
+                + 'showNavigation: true, showRuler: true, genome: "hg19", locus: "' + hugoSymbol + '", tracks: [  { url: "api/copynumbersegments", indexed: false, name: "Segmented CN", type:"seg", json: true, method: "POST", ';
+            var bodyContent2 = 'cancerStudyId: "' + cancerStudyId + '", hugoSymbol: "' + hugoSymbol +'", sampleIds: "' + sampleIds + '"},{name: "Genes", url: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg19/genes/gencode.v18.collapsed.bed", order: Number.MAX_VALUE,  displayMode: "EXPANDED"}]};igv.createBrowser(div, options);});<\/script><\/body>';    
+            var fullContent = headerContent + bodyContent1 + bodyContent2;
+            return fullContent;
+        }
+        
+        $("#igv-result-tab").click(function(){
+            var genes = window.QuerySession.getQueryGenes();
```

Comment:
I don't think so. User need to input gene to be able to get to the query page. But it is good to add a condition check just in case.